### PR TITLE
Simplify simple kmod

### PR DESCRIPTION
**Pull request by zvonkok**
Added kABI whitelist checking
Simplified the BuildConfig for simple-kmod.
Added a color module for pretty printing special resources. 
Simplified the log output more descriptive and concise. 

simple-kmod will be our test vehicle and POC for all things related to driver-containers on OpenShift.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,7 +100,7 @@ generate: controller-gen
 
 # Build the docker image
 local-image-build: test
-	podman build . -t ${IMG}
+	podman build --no-cache . -t ${IMG}
```
Comment (jmencak) [inline Makefile:103]: Is there a specific reason for the  `no-cache` ?

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -5,4 +5,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: quay.io/openshift-psap/special-resource-operator
-  newTag: monitoring
+  newTag: kvc-enhancement
```
Comment (jmencak) [inline config/manager/kustomization.yaml:8]: Why's this needed?

File: config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml [hunk 3]
```diff
@@ -47,43 +47,72 @@ data:
...
-    UNAME=$(uname -r)
+    kernel_version=$(uname -r)
 
-    echo "Enabling RHOCP and EUS RPM repos..."
-    yum config-manager --set-enabled rhocp-{{.ClusterVersionMajorMinor}}-for-rhel-8-x86_64-rpms || true
-    if ! yum makecache  --releasever={{.OperatingSystemDecimal}}; then 
-      yum config-manager --set-disabled rhocp-{{.ClusterVersionMajorMinor}}-for-rhel-8-x86_64-rpms 
-    fi 
+    # Adding the full name here so we can use one function 
+    # to install from EUS, base and MOC (machine-os-content)
```
Comment (jmencak) [inline config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml]: nit
s/base and MOC (machine-os-content)/base and machine-os-content (MOC)/  ?

**Review by jmencak — COMMENTED** [review]
Haven't tested, only briefly looked at this.  This looks as mostly only cosmetic changes.

File: Makefile (hunk 1 above)
Comment (zvonkok) [reply Makefile:103]: I had trouble building containers with minuscule changes that were cached and not invalidated.

File: config/manager/kustomization.yaml (hunk 2 above)
Comment (zvonkok) [reply config/manager/kustomization.yaml:8]: This is coming from `kustomize` this can be ignored. A `make deploy` will overwrite it anyway.

**Review by ArangoGutierrez — APPROVED** [review]
Looks good to me, deferring to @dagrayvid for final approval

**Comment by dagrayvid** [comment]
The changes look good and I was able to deploy from this branch and create the simple-kmod specialresource and verify that the simple-kmod is loaded by the systemd service.

/lgtm

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

File: config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml (hunk 3 above)
Comment (zvonkok) [reply config/recipes/driver-container-base/manifests/0000-driver-container-base-build.yaml]: Done

**Pull request merged**